File: app/src/main/java/com/parabola/newtone/mvp/view/FolderView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.OneExecution;

@AddToEndSingle
public interface FolderView extends MvpView {

    void setFolderPath(String folderPath);

    void refreshTracks(List<Track> tracks);
    void setItemViewSettings(TrackItemView viewSettings);
    void setItemDividerShowing(boolean showed);
    void setSectionShowing(boolean enable);

    void setCurrentTrack(int trackId);

    @OneExecution
    void showToast(String message);
}

File: app/src/main/java/com/parabola/newtone/presentation/folder/FolderView.kt
package com.parabola.newtone.presentation.folder

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface FolderView : MvpView {
    fun setFolderPath(folderPath: String)

    fun refreshTracks(tracks: List<Track>)

    fun setItemViewSettings(viewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)
    fun setSectionShowing(enable: Boolean)

    fun setCurrentTrack(trackId: Int)
}
